ivy/__init__.py

```diff
@@ -192,6 +192,15 @@ def can_cast(self, to):
         return can_cast(self, to)
 
 
+def shape_casting_helper(ivy_shape, other):
```

Comment:
Can this be implemented as a private static method of the `Shape` class instead?